src/meta-srv/src/election/mysql.rs

```diff
@@ -0,0 +1,1553 @@
+    store_key_prefix: String,
+    candidate_lease_ttl_secs: u64,
+    sql_set: ElectionSqlSet,
+}
+
+impl MySqlElection {
+    pub async fn with_mysql_client(
+        leader_value: String,
+        mut client: sqlx::MySqlConnection,
+        store_key_prefix: String,
+        candidate_lease_ttl_secs: u64,
+        table_name: &str,
+    ) -> Result<ElectionRef> {
+        let sql_factory = ElectionSqlFactory::new(table_name);
+        sqlx::query(&sql_factory.create_table_sql())
+            .execute(&mut client)
+            .await
+            .context(MySqlExecutionSnafu)?;
+        // Set idle session timeout to IDLE_SESSION_TIMEOUT to avoid dead lock.
+        sqlx::query(&sql_factory.set_idle_session_timeout_sql())
+            .execute(&mut client)
+            .await
+            .context(MySqlExecutionSnafu)?;
+        // Set lock wait timeout to LOCK_WAIT_TIMEOUT to avoid waiting too long.
+        sqlx::query(&sql_factory.set_lock_wait_timeout_sql())
+            .execute(&mut client)
+            .await
+            .context(MySqlExecutionSnafu)?;
+        // Insert at least one row for `SELECT * FOR UPDATE` to work.
+        sqlx::query(&sql_factory.insert_once())
+            .execute(&mut client)
+            .await
+            .context(MySqlExecutionSnafu)?;
+        let tx = listen_leader_change(leader_value.clone());
+        Ok(Arc::new(Self {
+            leader_value,
+            client: Mutex::new(client),
+            is_leader: AtomicBool::new(false),
+            leader_infancy: AtomicBool::new(false),
+            leader_watcher: tx,
+            store_key_prefix,
+            candidate_lease_ttl_secs,
+            sql_set: sql_factory.build(),
+        }))
+    }
+
+    fn election_key(&self) -> String {
+        format!("{}{}", self.store_key_prefix, ELECTION_KEY)
+    }
+
+    fn candidate_root(&self) -> String {
+        format!("{}{}", self.store_key_prefix, CANDIDATES_ROOT)
+    }
+
+    fn candidate_key(&self) -> String {
+        format!("{}{}", self.candidate_root(), self.leader_value)
+    }
+}
+
+#[async_trait::async_trait]
+impl Election for MySqlElection {
+    type Leader = LeaderValue;
+
+    fn is_leader(&self) -> bool {
+        self.is_leader.load(Ordering::Relaxed)
+    }
+
+    fn in_leader_infancy(&self) -> bool {
+        self.leader_infancy
+            .compare_exchange(true, false, Ordering::Relaxed, Ordering::Relaxed)
+            .is_ok()
+    }
+
+    /// TODO(CookiePie): Split the candidate registration and keep alive logic into separate methods, so that upper layers can call them separately.
+    async fn register_candidate(&self, node_info: &MetasrvNodeInfo) -> Result<()> {
+        let key = self.candidate_key();
+        let node_info =
+            serde_json::to_string(node_info).with_context(|_| SerializeToJsonSnafu {
+                input: format!("{node_info:?}"),
+            })?;
+
+        info!("Registering candidate now, getting client");
```

Comment:
`info!` might be too verbose
```suggestion
        debug!("Registering candidate now, getting client");
```